src/pudl/analysis/plant_parts_eia.py

```diff
@@ -290,6 +294,7 @@
     "energy_source_code_1",
     "prime_mover_code",
     "ferc_acct_name",
+    "operating_year",
```

Comment:
Honestly I got all sorts of confused diving into the `AddConsistentAttributes` class but I think it's fine to just add `operating_year` to this list? This is what fixed the original bug.

Reply:
Yes, you can just add this!

Reply:
I added an assert in `MakePlantPart` that ensures that all ID columns are in each part dataframe